bigquery/tests/system.py

```diff
@@ -1235,6 +1235,30 @@ def test_query_future(self):
         row_tuples = [r.values() for r in iterator]
         self.assertEqual(row_tuples, [(1,)])
 
+    def test_query_iter(self):
+        import types
+        query_job = Config.CLIENT.query('SELECT 1')
+        self.assertIsInstance(iter(query_job), types.GeneratorType)
+        row_tuples = [r.values() for r in query_job]
+        self.assertEqual(row_tuples, [(1,)])
+
+    def test_query_to_dataframe(self):
+        import pandas as pd
```

Comment:
It doesn't seem like this needs to be a system test, since (as above), it is just wrapping `.result()` which is presumably already system-tested.

----

If you insist, I'd prefer the system tests to be runnable without `pandas`. To make this possible you could:

```python
try:
    import pandas
except ImportError:
    pandas = None
```

and then use

```python
@unittest.skipIf(pandas is None, 'Requires `pandas`')
def test_query_to_dataframe(self):
    ...
```
----

In general, our style is "import at run time in unit tests" and "import at import time" in system tests. The idea being that you will run unit tests **a lot** and don't want a silly error (e.g. a syntax error, which would fail an import) to fail the entire unit test suite at **import time**.

Reply:
Removed the system test in 823b686. Fixed the imports and skipped the tests in 70d6bd9